cl/citations/management/commands/westlaw_citation_merger.py

```diff
@@ -0,0 +1,354 @@
+                    )
+                else:
+                    logger.warning(
+                        f'Cluster {citation_cluster_results.first().pk} already have both citations "{citation}" and "{parallel_citation}"'
+                    )
+            elif results_count > 1 and parallel_results_count == 0:
+                # We got more than one result using the citation, we can
+                # try to find the case name using csv data and try to
+                # match it with the correct case
+
+                # Check if case name from csv is exactly the same in
+                # case_names list and get index, this is just a workaround
+                # to use as much of westlaw data as possible, we could try
+                # to implement something more complex
+                indexes_found = find_indexes(list(results_names), case_name)
+                if len(indexes_found) == 1:
+                    # Add parallel citation to cluster obtained from list
+                    case_id = results_ids[indexes_found[0]]
+                    add_citation(parallel_citation, case_id, debug)
+                elif len(indexes_found) > 1:
+                    logger.warning(
+                        f'Multiple results for case: "{case_name}" and citation: "{citation}"'
+                    )
+
+            elif results_count == 0 and parallel_results_count > 1:
+                # We got more than one result using parallel citation,
+                # we can try to find the case name using csv data and
+                # try to match it with the correct case
+
+                # Check if the csv case name is exactly the same in
+                # the case_names list and get the index
+                indexes_found = find_indexes(
+                    list(parallel_results_names), case_name
+                )
+                if len(indexes_found) == 1:
+                    case_id = parallel_results_ids[indexes_found[0]]
+                    # Add citation to parallel citation cluster
+                    add_citation(citation, case_id, debug)
+                elif len(indexes_found) > 1:
+                    # Many case names have the same names, it's hard
+                    # to figure out where citation belongs to
+                    logger.warning(
+                        f'Multiple results for case: "{case_name}" and '
+                        f'parallel citation: "{parallel_citation}"'
+                    )
+
+            elif results_count == 0 and parallel_results_count == 0:
+                # No match for both citations
+                logger.warning(
+                    f'No results for case: "{case_name}" with '
+                    f'citation: "{citation}" and parallel citation: '
+                    f'"{parallel_citation}"'
+                )
```

Comment:
yes, i would leave that as last resort, because in the tests i did, i almost got the same results count filtering by court, with date i found many times that we have a different dates than the ones from the csv, i'll give it a try and let you know when is ready